core/src/main/scala/com/gu/tableversions/core/InMemoryTableVersions.scala

```diff
@@ -17,95 +13,63 @@ import com.gu.tableversions.core.util.RichRef._
 class InMemoryTableVersions[F[_]] private (allUpdates: Ref[F, TableUpdates])(implicit F: Sync[F])
     extends TableVersions[F] {
 
-  override def init(
-      table: TableName,
-      isSnapshot: Boolean,
-      userId: UserId,
-      message: UpdateMessage,
-      timestamp: Instant): F[Unit] =
-    allUpdates.update { prev =>
-      if (prev.contains(table)) prev
-      else
-        prev + (table -> List(
-          TableUpdate(
-            userId,
-            message,
-            timestamp = timestamp,
-            operations = List(InitTable(table, isSnapshot))
-          )))
+  override def commit(table: TableName, update: TableVersions.TableUpdate): F[Unit] = {
+    val applyUpdate: TableUpdates => Either[Exception, TableUpdates] = { currentTableUpdates =>
+      if (currentTableUpdates.contains(table)) {
+        val updated = currentTableUpdates + (table -> TableState(
+          currentVersion = update.header.id,
+          updates = currentTableUpdates(table).updates :+ update))
+        Right(updated)
+      } else
+        Left(unknownTableError(table))
     }
 
-  override def currentVersion(table: TableName): F[TableVersion] =
-    // Derive current version of a table by folding over the history of changes
-    for {
-      allTableUpdates <- allUpdates.get
-      tableUpdates <- allTableUpdates
-        .get(table)
-        .fold(F.raiseError[List[TableUpdate]](new Exception(s"Table '$table' not found")))(F.pure)
-      operations = tableUpdates.flatMap(_.operations)
-    } yield
-      if (isSnapShot(operations))
-        InMemoryTableVersions.latestSnapshotTableVersion(operations)
-      else
-        InMemoryTableVersions.applyPartitionUpdates(PartitionedTableVersion(Map.empty))(operations)
-
-  private def isSnapShot(operations: List[TableOperation]) = operations match {
-    case InitTable(_, isSnapshot) :: _ => isSnapshot
-    case _                             => throw new IllegalArgumentException("First operation should be InitTable")
+    allUpdates.modifyEither(applyUpdate).void
```

Comment:
naive question. What purpose does `.void` do seeing `modifyEither(...)` return type is `F[Unit]` ?

Reply:
Not naive - that is indeed unneccessary! 😊